### PR TITLE
Fix env setting for tools/lib_config

### PR DESCRIPTION
**Pull request by 2maz**
... to setup the PYTHONPATH for this package

### CODE REVIEW

File: libs.autobuild [hunk 1]
```diff
@@ -8,7 +8,10 @@ cmake_package 'tools/orogen_model_exporter' do |pkg|
     pkg.env_add_path 'RUBYLIB', File.join(pkg.srcdir, 'lib')
     pkg.env_add_path 'OROGEN_PLUGIN_PATH', File.join(pkg.prefix, 'share','orogen','plugins')
 end
-cmake_package 'tools/lib_config'
+cmake_package 'tools/lib_config' do |pkg|
+    bin,_,_ = Rock.activate_python_path(pkg)
+    pkg.define "PYTHON_EXECUTABLE", bin if bin
```
Comment (maltewi) [inline libs.autobuild:13]: this produces this message:

```
  tools/lib_config: changed value of PYTHON_EXECUTABLE from /media/wirkus/Data/development/artemis-again/install/bin/python to /usr/bin/python3
```
is this intended? 

The file `.../install/bin/python` looks like this:
```
#! /bin/sh
exec /usr/bin/python3 "$@"
```

Comment (2maz) [reply libs.autobuild:13]: Yes, this is correct - install/bin/python is generated by the python helper in rock.core and is based on a user's selection of the python binary

Comment (maltewi) [reply libs.autobuild:13]: Ok, but why and how does `Rock.activate_python_path` decide to switch to a different python executable? (effectively it's the same executable in my case).

Or, more relevant from the scope of the package at hand: what is the definition of PYTHON_EXECUTABLE good for from the perspective of `tools/lib_config`?
I understand that its important to activate the PYTHONPATH for the package (the line above), but why does the package have to care about this PYTHON_EXECUTABLE variable?

Comment (2maz) [reply libs.autobuild:13]: Rock.activate_python_path queries the information about the python version to use from the autoproj user: if you do autoproj reconfigure, you will see the respective question during the configuration. So Rock.activate_python_path is only validating the user selection and updates the shims (install/bin/python and install/bin/pip) accordingly.

Now, to ensure that the user has to do the selection only once (autoproj reconfigure) and using only one code path for the identification of the executable, the PYTHON_EXECUTABLE is forwarded via pkg.define. This mechanism is for consistency and avoids in a better case a redundant and in worse case an inconsistent custom detection of your active python executable in your cmake.

Comment (maltewi) [reply libs.autobuild:13]: The CMake uses PythonInterp to determine the python version: https://github.com/rock-cpp/lib_config/blob/fd481b2719edcfd72c0e771dee4b4f8560d6ca8a/bindings/python/CMakeLists.txt#L1

Does this match together with the method Rock.activate_python_path uses to query the information, or is there a danger that two different versions are detected?

Comment (2maz) [reply libs.autobuild:13]: I do not known the details on PythonInterp - but I guess that is picking the python version that is found in PATH. Which is why you had the shims in install/bin/python set and not /usr/bin/python3. In principle fine here, but redundant. Note, that if you want to identify the python version, one can as well forward the python version:

```
    bin, version, sitelib_path = Rock.activate_python_path(pkg)
    pkg.define "PYTHON_EXECUTABLE", bin if bin
    pkg.define "PYTHON_VERSION", version if version
```

**Pull request merged**